kubernetes/schema_pod_spec.go

```diff
@@ -715,15 +726,12 @@ func volumeSchema(isUpdatable bool) *schema.Resource {
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
 				"medium": {
-					Type:        schema.TypeString,
-					Description: `What type of storage medium should back this directory. The default is "" which means to use the node's default medium. Must be an empty string (default) or Memory. More info: https://kubernetes.io/docs/concepts/storage/volumes#emptydir`,
-					Optional:    true,
-					Default:     "",
-					ForceNew:    !isUpdatable,
-					ValidateFunc: validation.StringInSlice([]string{
-						string(corev1.StorageMediumDefault),
-						string(corev1.StorageMediumMemory),
-					}, false),
+					Type:         schema.TypeString,
+					Description:  fmt.Sprintf(`What type of storage medium should back this directory. The default is "" which means to use the node's default medium. Must be one of %q. More info: https://kubernetes.io/docs/concepts/storage/volumes#emptydir`, validEmptyDirMediums),
+					Optional:     true,
+					Default:      "",
+					ForceNew:     !isUpdatable,
+					ValidateFunc: validation.StringInSlice(validEmptyDirMediums, false),
```

Comment:
You'll want to also add a Test that covers the HugePage values. the test can be named `TestAccKubernetesPodV1_with_empty_dir_volume_huge_page` which would be created in `resource_kubernetes_pod_v1.go`.

You can use other tests in that file as reference when creating the test for HugePage. Be sure to run the test on your machine.

Reply:
Done, added TestAccKubernetesDeploymentV1_with_empty_dir_huge_page.

I went with a deployment rather than a pod so I could set `replicas = 0`, since my locally run K8S cluster (I used `kind`) didn't have huge page support that I was easily able to figure out, and I didn't want to put that onus on others either.

Reply:
Thanks! I'll be testing it locally first before approving